d2-schemas/src/main/pegasus/com/linkedin/d2/D2QuarantineProperties.pdl

```diff
@@ -0,0 +1,21 @@
+namespace com.linkedin.d2
+
+/**
+ * Properties for D2 quarantine.
+ *
+ * Uhealthy hosts can be placed in quarantine, in which a predefined method is used
+ * to check endpoint health without routing real request traffic to them.
+ */
+record D2QuarantineProperties {
+
+  /**
+   * The max percentage of hosts in the cluster that can be quarantined at the same time.
+   * Values are from 0.0 - 1.0. A value of 0.0 means quarantine is disabled.
+   */
+  quarantineMaxPercent: optional double
+
+  /**
+   * Specifies the method used for health checking. Format: <METHOD>:<FULL_PATH>. Defaults to OPTIONS method.
+   */
+  quarantineMethod: optional string
```

Comment:
Could this possibly be an enum for the method, combined with another field for the path? Rather than just a string.

Reply:
Actually, thanks for pointing this out. I will drop this field altogether as I realized now that it is not necessary.

Reply:
Yes, I can separate it out into 2 fields:
1. healthCheckMethod: optional enum (Do you know of a PDL for http methods?)
2. healthCheckPath: optional string

Reply:
Looks good, thanks